### DISCUSSION TITLE
TuYa TRV not updating temperature unless I change offset

### DISCUSSION THREAD

**Discussion by elcriuz** [Koenkk/zigbee2mqtt#12075] [Devices]
I have a TuYa TRV-02 connected to my Z2M and apparently everything works as expected, except that local_temperature is not updating at all. When I nudge local_temperature_calibration in any direction, local_temperature suddenly updates. How can I fix this?

**Comment by lux73** [comment]
https://github.com/Koenkk/zigbee2mqtt/issues/10547

it seems your TRV's need an Firmware Upgrade via Tuya-Hub

**Comment by ices-dev** [comment]
Yesterday I received my Tuya TVR and I have the same issue. I don’t have a tuya hub. Any other ideas?
